src/lib/location.ts

```diff
@@ -0,0 +1,79 @@
+import { Database } from 'sqlite';
+import { geocoder } from './geocoder';
+
+export type ResolvedLocation = {
+  state: string;
+  zcta: string;
+  city: string;
+  county: string;
+  tractGeoid?: string;
+};
+
+/**
+ * Geocodio returns an "accuracy_type" value that indicates how granular of a
+ * match it found. (Really it should be "precision", not "accuracy".) These
+ * types are imprecise enough, in practice, that we can't trust the geocoded
+ * census tract.
+ */
+const GEOCODIO_LOW_PRECISION = new Set([
+  'street_center',
+  'place',
+  'county',
+  'state',
+]);
```

Comment:
This is an editorial decision. The other `accuracy_type` values seem to be precise enough that they're not going to possibly span multiple census tracts. See: https://www.geocod.io/docs/#accuracy-score